src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationProviders.xml

```diff
@@ -2045,6 +2045,25 @@
     </Environment>
   </Provider>
 
+  <!--
+                                                   ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
+                                                   ██ ███ █ █▀▄████▄ ▄██ ▄▄▀██
+                                                   ███ █ ██ ▄▀██████ ███ ██ ██
+                                                   ███▄▀▄██ ██ ████▀ ▀██ ▀▀ ██
+                                                   ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
+  -->
+
+  <Provider Name="VkId" Id="9bf89c19-401b-4076-893e-a4136e719432" 
+            Documentation="https://id.vk.com/about/business/go/docs/en/vkid/latest/oauth-vk">
+    <Environment Issuer="https://id.vk.com/">
+      <Configuration AuthorizationEndpoint="https://id.vk.com/authorize"
+                     TokenEndpoint="https://id.vk.com/oauth2/auth"
+                     UserInfoEndpoint="https://id.vk.com/oauth2/user_info">
+        <CodeChallengeMethod Value="S256" />
```

Comment:
It seems that both providers support refresh tokens so we should definitely add that:

```xml
<GrantType Value="authorization_code" />
<GrantType Value="refresh_token" />
```

Please give it a try using the console sandbox to ensure the token refreshing works.